### DISCUSSION TITLE
Have tree-style package.json scripts been discussed?

### DISCUSSION THREAD

**Discussion by NashJames** [pnpm/pnpm#6745] [Ideas]
The main idea is to tidy the set of package.json scripts, but it may also allow use of something like `pnpm format` to run both or `pnpm format:eslint` to run one. E.g:
```json
"scripts": {
    "dev": "next dev",
    "start": "next start",
    "build": "tsc && next build",
    "format": {
        "eslint": "next lint --fix",
        "prettier": "prettier --write .",
    },
}
```

Couldn't find any mention on it. Doesn't feel like a particularly original idea so wondering if there's an answer somewhere already?

**Comment by thynson** [comment]
That would breaks the compatibility with other package managers. 
And the purpose in your example can be done simply with `"format": "next lint --fix && prettier --write ."; while if it became too long, a dedicated bash script is the best way for it.

**Comment by NashJames** [comment]
Cross-compatibility with other package managers isn't an issue surely? They can either add the functionality or you only choose to structure your package.json this way if you intend to use pnpm (they can always refactor it if they change package managers).

And yes I'm well aware you can do both of those options. But they don't leave your setup very tidy. Especially in larger projects like monorepos where you can end up with 20-40 scripts. Many of which might just exist to `&&` other scripts so they are more easily visible on smaller laptop screens.
